[DOCS] Updates add data content (#81093)
elastic/kibana: docs/setup/connect-to-elasticsearch.asciidoc

[[connect-to-elasticsearch]]

> == Add data

Reviewer: It is confusing that this line and line 18 are both `Add data`.

++++
<titleabbrev>Add data</titleabbrev>
++++

To start working with your data in {kib}, use one of the many ingest options,
available from the home page.
You can collect data from an app or service
or upload a file that contains your data.  If you're not ready to use your own data,
add a sample data set and give {kib} a test drive.

[role="screenshot"]
image::images/add-data-home.png[Built-in options for adding data to Kibana:  Add data, Add Elastic Agent, Upload a file]

[float]
[[add-data-tutorial-kibana]]
===  Add data

Want to ingest logs, metrics, security, or application data?
Install and configure a Beats data shipper or other module to periodically collect the data
and send it to {es}. You can then use the pre-built dashboards to explore and analyze the data.

[role="screenshot"]
image::images/add-data-tutorials.png[Add Data tutorials]

[discrete]
=== Add Elastic Agent

> beta[] *Elastic Agent* is sneak peek at the next generation of

Reviewer: ```suggestion
beta[] *Elastic Agent* is a sneak peek at the next generation of
```

data integration modules, offering
a centralized way to set up your integrations.
With *Fleet*, you can add
and manage integrations for popular services and platforms, providing
an easy way to collect your data. The integrations
ship with dashboards and visualizations,

> so you quickly get insights on your data.

Reviewer: ```suggestion
so you can quickly get insights into your data.
```

To get started, refer to
{ingest-guide}/ingest-management-getting-started.html[Quick start: Get logs and metrics into the Elastic Stack].

[role="screenshot"]
image::images/add-data-fleet.png[Add data using Fleet]

[discrete]
[[upload-data-kibana]]
=== Upload a file

experimental[] If your data is in a CSV, JSON, or log file, you can upload it using the File
Data Visualizer. You can upload a file up to 100 MB. This value is configurable up to 1 GB in
<<kibana-ml-settings, Advanced Settings>>. To upload a file with geospatial data,

> see <<import-geospatial-data, Import geospatial data>>.

Reviewer: ```suggestion
refer to <<import-geospatial-data, Import geospatial data>>.
```

[role="screenshot"]
image::images/add-data-fv.png[File Data Visualizer]

NOTE: This feature is not intended for use as part of a
repeated production process, but rather for the initial exploration of your data.

[discrete]
=== Additional options for loading your data

If the {kib} ingest options don't work for you, you can index your
data into Elasticsearch with {ref}/getting-started-index.html[REST APIs]
or https://www.elastic.co/guide/en/elasticsearch/client/index.html[client libraries].
After you add your data, you're required to create an <<index-patterns,index pattern>> to tell
{kib} where to find the data.

> * To add data for Elastic Observability, see {observability-guide}/add-observability-data.html[Send data to Elasticsearch].

Reviewer: ```suggestion
* To add data for Elastic Observability, refer to {observability-guide}/add-observability-data.html[Send data to Elasticsearch].
```

> * To add data for Elastic Security, see https://www.elastic.co/guide/en/security/current/ingest-data.html[Ingest data to Elastic Security].

Reviewer: ```suggestion
* To add data for Elastic Security, refer to https://www.elastic.co/guide/en/security/current/ingest-data.html[Ingest data to Elastic Security].
```

[[connect-to-elasticsearch]]
== Add data to {kib}
++++
<titleabbrev>Add data</titleabbrev>
++++

> The fastest way to add data to {kib} is to use one of the built-in options.

Reviewer 2: One comment about "add data to Kibana". This statement is a little technically misleading. Kibana does not store data so you never add data to kibana. You add data to Elasticsearch or the Elastic stack. Sometimes you add data by using kibana like with sample data or file upload.

Reviewer 3: I agree.

> The fastest way to add data to {kib} is to use one of the built-in options.

Reviewer 4: Thoughts on replacing "use one of our built in options" with "one of our many ingest options"?

You can collect data from an app or service
or upload a file that contains your data.  If you're not ready to use your own data,
add a sample data set and give {kib} a test drive.
The built-in options are available on the home page.

[role="screenshot"]
image::images/add-data-home.png[Built-in options for adding data to Kibana:  Add data, Add Elastic Agent, Upload a file]

[float]
[[add-data-tutorial-kibana]]
===  Add data

Want to ingest logs, metrics, security, or application data?
Install and configure a Beats data shipper or other module to periodically collect the data
and send it to {es}. You can then use the pre-built dashboards to explore and analyze the data.

[role="screenshot"]
image::images/add-data-tutorials.png[Add Data tutorials]

[discrete]

> === Add Elastic Agent

Reviewer 4: These two things are very similar, integrations with Agent + Fleet are the next generation of modules. I wonder if we can spin this in a way with "get a sneak peak of where we're going with.." or something along those lines.

beta[] *Elastic Agent* is a centralized way to set up your data integrations.

> With the *Fleet* UI, you can add

Reviewer 4: Should we just say with "Fleet" instead of "Fleet UI"?

and manage integrations for popular services and platforms, providing
an easy way to collect your data. The integrations
ship with dashboards and visualizations,
so you quickly get insights on your data.

To get started, refer to
{ingest-guide}/ingest-management-getting-started.html[Quick start: Get logs and metrics into the Elastic Stack].

[role="screenshot"]
image::images/add-data-fleet.png[Add data using Fleet]

> NOTE: This feature is not intended for use as part of a

Reviewer 3: Not sure I like this note because it's a little misleading. Eventually it *will* be the main way to ingest data, but right now, it's not recommended in a production environment. Maybe qualify it by saying, "The beta release of this feature...."

repeated production process, but rather for the initial exploration of your data.

[discrete]
[[upload-data-kibana]]
=== Upload a file

experimental[] If your data is in a CSV, JSON, or log file, you can upload it using the File
Data Visualizer. You can upload a file up to 100 MB. This value is configurable up to 1 GB in
<<kibana-ml-settings, Advanced Settings>>.

[role="screenshot"]
image::images/add-data-fv.png[File Data Visualizer]

> For geospatial data, you can use the File Data Visualizer or the

Reviewer 2: I don't think you need to call out File Data Visualizer or GeoJSon Uploader. How about just changing to "See <<import-geospatial-data, Import geospatial data>> for uploading files with geospatial data."

Reviewer 4: ++

GeoJSon Uploader, depending on the format of your data.  See
<<import-geospatial-data, Import geospatial data>> for details.

NOTE: This feature is not intended for use as part of a
repeated production process, but rather for the initial exploration of your data.

[discrete]

> === Custom options for loading your data

Reviewer 4: Thoughts on using "Additional options for loading your data"? or "Options for loading your custom data"

> If the built-in options don't work for you, you can index your

Reviewer 4: Similar to my comment above, thoughts on replacing "If the built-in options don't work for you" with "If one of our multiple ingest options don't work for you"?

data into Elasticsearch with {ref}/getting-started-index.html[REST APIs]
or https://www.elastic.co/guide/en/elasticsearch/client/index.html[client libraries].
After you add your data, you're required to create an <<index-patterns,index pattern>> to tell
{kib} where to find the data.

> * To add data for Elastic Observability, see {observability-guide}/add-observability-data.html[Send data to Elasticsearch].

Reviewer 4: Probably another discussion, but a common naming convention and framework for ingestion "How to's" might make it easier to consume for our end users.

[[connect-to-elasticsearch]]
== Add data to {kib}

The fastet way to add data to {kib} is to use one of the built-in options,
available from the home page. You can:

* Add data from popular apps and services

> * Add data using an Elastic Agent

Reviewer 3: Not exactly clear how line 7 is different from line 8 (Elastic Agent is the new way of adding data from popular apps and services, no?)

Reviewer 3: I'd also indicate here that this is beta

* Upload a file

[role="screenshot"]
image::images/add-data-home.png[Built-in options for adding data to Kibana]

If you're not ready to use your own data, you can add a <<get-started, sample data set>>
to give {kib} a test drive.

[float]
[[add-data-tutorial-kibana]]
===  Add data tutorials

These tutorials guide you through installing and configuring a
Beats data shipper to collect metrics, logs, security events, and application data and send it to {es}.
You can then use the pre-built dashboards to explore and analyze the data.

If a tutorial doesn’t exist for your data, go to the {beats-ref}/beats-reference.html[Beats overview]
to learn about other data shippers in the Beats family.

[role="screenshot"]
image::images/add-data-tutorials.png[Add Data tutorials]

> 

Reviewer 3: Might be worth stating explicitly somewhere that Elastic Agent is an alternative to installing and managing Beats individually. I think there will be some confusion for users because Elastic Agent installs and manages Beats under the hood. We really want that to be an implementation detail that most users don't need to know about, though.

[discrete]
[[add-data-fleet-kibana]]
===  Add an Elatic Agent

beta[]

> Fleet is a single, unified way to collect logs and metrics from systems

Reviewer 3: This description blurs the lines between Fleet and agent. Agent is how you collect logs/metrics. Fleet is the UI you use to add and manage agents. You might want to look at the Fleet overview (thought honestly I want to rework that because it's a bit disjointed). https://www.elastic.co/guide/en/ingest-management/7.10/fleet-overview.html

> and services across your organization. You don't have to install multiple

Reviewer 3: "You don't have to" makes it sound like this is optional.

Maybe: "Instead of installing multiple agents on each host you want to monitor, you install a single agent that manages data collection, making it easier to deploy policies across your infrastructure."

(Kind of wordy and I'm not sure users will understand what a policy is at this point, but hopefully you see my point.)
